### ISSUE TITLE
Docker service became unhealthy. It's container was killed and was never rescheduled while still being visible in `docker ps` (rpc error: code = 2 desc = containerd: container not found)

### ISSUE THREAD

**Issue by bzak9LD** [moby/moby#36945]
**Description**

We've setup a swarm cluster, with 10-20 services spanned across 3 nodes. 
After some time one service became unhealthy. It's container was still visible in `docker ps` but we couldn't exec to it/connect to it and `docker inspect`'s health was of the following form:
```
"Health": {
                "Status": "unhealthy",
                "FailingStreak": 30,
                "Log": [
                    {
                        "Start": "2018-04-24T01:59:16.951628307Z",
                        "End": "2018-04-24T01:59:16.956321731Z",
                        "ExitCode": -1,
                        "Output": "rpc error: code = 2 desc = containerd: container not found"
                    },
...
```
Service remained in unhealthy state, it didn't schedule new containers (for 10+ hours).

Restarting docker daemon workarounded this problem.

**Steps to reproduce the issue:**
This issue is nondeterministic and happened only once so far.

**Describe the results you received:**
Docker service's container "failed" and swarm never rescheduled it.

**Describe the results you expected:**
Docker swarm will eventually reschedule failed container. 

**Additional information you deem important (e.g. issue happens only occasionally):**

I've collected some relevant parts of the syslog and aranged them into this timeline:
Above mentioned container's name is _0e9a27f_.

1. **01:55:01** Healthchecks of containerd start to fail:
```
Apr 24 01:55:01 localhost dockerd: time="2018-04-24T01:55:01.642401003Z" level=debug msg="libcontainerd: containerd health check returned error: rpc error: code = 4 desc = context deadline exceeded"
Apr 24 01:55:08 localhost dockerd: time="2018-04-24T01:55:08.326048270Z" level=error msg="libcontainerd: failed to receive event from containerd: rpc error: code = 13 desc = transport is closing"
Apr 24 01:55:10 localhost dockerd: time="2018-04-24T01:55:09.560606698Z" level=warning msg="failed to retrieve containerd version: rpc error: code = 14 desc = grpc: the connection is unavailable"
```
2. **01:55:03** Healthcheck for container start to fail:
```
Apr 24 01:54:38 localhost dockerd: time="2018-04-24T01:54:38.054017530Z" level=debug msg="Health check for container 0e9a27fd4092a868cafc98f15c205f5239b0319825084c14b8c027600d789a3e done (exitCode=0)"
Apr 24 01:54:48 localhost dockerd: time="2018-04-24T01:54:48.151670662Z" level=debug msg="Health check for container 0e9a27fd4092a868cafc98f15c205f5239b0319825084c14b8c027600d789a3e done (exitCode=0)"
Apr 24 01:55:03 localhost dockerd: time="2018-04-24T01:55:03.152111727Z" level=debug msg="Health check for container 0e9a27fd4092a868cafc98f15c205f5239b0319825084c14b8c027600d789a3e taking too long"
Apr 24 01:55:03 localhost dockerd: time="2018-04-24T01:55:03.158457456Z" level=warning msg="Health check for container 0e9a27fd4092a868cafc98f15c205f5239b0319825084c14b8c027600d789a3e error: rpc error: code = 4 desc = context deadline exceeded"
Apr 24 01:55:13 localhost dockerd: time="2018-04-24T01:55:13.165431417Z" level=warning msg="Health check for container 0e9a27fd4092a868cafc98f15c205f5239b0319825084c14b8c027600d789a3e error: rpc error: code = 14 desc = grpc: the connection is unavailable"
Apr 24 01:55:23 localhost dockerd: time="2018-04-24T01:55:23.174704046Z" level=warning msg="Health check for container 0e9a27fd4092a868cafc98f15c205f5239b0319825084c14b8c027600d789a3e error: rpc error: code = 2 desc = containerd: container not found"
```
3. **01:55:11** Containerd gets restarted multiple times with no success:
```
Apr 24 01:55:11 localhost dockerd: time="2018-04-24T01:55:11.619396647Z" level=info msg="libcontainerd: new containerd process, pid: 29913"
Apr 24 01:55:11 localhost dockerd: time="2018-04-24T01:55:11.619569169Z" level=debug msg="libcontainerd: containerd health check returned error: rpc error: code = 14 desc = grpc: the connection is unavailable"
Apr 24 01:55:11 localhost dockerd: time="2018-04-24T01:55:11.642017602Z" level=debug msg="libcontainerd: containerd health check returned error: rpc error: code = 14 desc = grpc: the connection is unavailable"
Apr 24 01:55:12 localhost dockerd: time="2018-04-24T01:55:12.142010409Z" level=debug msg="libcontainerd: containerd health check returned error: rpc error: code = 14 desc = grpc: the connection is unavailable"
Apr 24 01:55:12 localhost dockerd: time="2018-04-24T01:55:12.147163209Z" level=info msg="libcontainerd: new containerd process, pid: 29937"
Apr 24 01:55:12 localhost dockerd: time="2018-04-24T01:55:12.642096224Z" level=debug msg="libcontainerd: containerd health check returned error: rpc error: code = 14 desc = grpc: the connection is unavailable"
Apr 24 01:55:13 localhost dockerd: time="2018-04-24T01:55:13.142005864Z" level=debug msg="libcontainerd: containerd health check returned error: rpc error: code = 14 desc = grpc: the connection is unavailable"
Apr 24 01:55:13 localhost dockerd: time="2018-04-24T01:55:13.642159235Z" level=debug msg="libcontainerd: containerd health check returned error: rpc error: code = 14 desc = grpc: the connection is unavailable"
Apr 24 01:55:13 localhost dockerd: time="2018-04-24T01:55:13.651120111Z" level=info msg="libcontainerd: new containerd process, pid: 29941"
Apr 24 01:55:14 localhost dockerd: time="2018-04-24T01:55:14.142011165Z" level=debug msg="libcontainerd: containerd health check returned error: rpc error: code = 14 desc = grpc: the connection is unavailable"
Apr 24 01:55:14 localhost dockerd: time="2018-04-24T01:55:14.642065331Z" level=debug msg="libcontainerd: containerd health check returned error: rpc error: code = 14 desc = grpc: the connection is unavailable"
Apr 24 01:55:15 localhost dockerd: time="2018-04-24T01:55:15.142078742Z" level=debug msg="libcontainerd: containerd health check returned error: rpc error: code = 14 desc = grpc: the connection is unavailable"
```
4. **01:55:15** Containerd starts but seems to have inconsistent state of the container:
```
Apr 24 01:55:15 localhost dockerd: time="2018-04-24T01:55:15.146541753Z" level=info msg="libcontainerd: new containerd process, pid: 29972"
Apr 24 01:55:15 localhost dockerd: time="2018-04-24T01:55:15.274845171Z" level=debug msg="containerd: grpc api on /var/run/docker/libcontainerd/docker-containerd.sock"
Apr 24 01:55:15 localhost dockerd: time="2018-04-24T01:55:15.280149667Z" level=debug msg="containerd: read past events" count=799
Apr 24 01:55:15 localhost dockerd: time="2018-04-24T01:55:15.444455384Z" level=warning msg="containerd: failed to load container,removing state directory." error="open /var/run/docker/libcontainerd/containerd/0e9a27fd4092a868cafc98f15c205f5239b0319825084c14b8c027600d789a3e/872a6ff349bdea5b4c4090ac3ab68b0d52154b4b9d770922fa37981d79871a5f/process.json: no such file or directory" id=0e9a27fd4092a868cafc98f15c205f5239b0319825084c14b8c027600d789a3e
```
5. **01:59:59** Docker reports it can't kill container:
```
Apr 24 01:59:59 localhost dockerd: time="2018-04-24T01:59:59.988439668Z" level=warning msg="container kill failed because of 'container not found' or 'no such process': Cannot kill container 0e9a27fd4092a868cafc98f15c205f5239b0319825084c14b8c027600d789a3e: rpc error: code = 2 desc = containerd: container not found"
```
6. **02:00:22** Docker does a _sigkill_ on the container:
```
Apr 24 02:00:22 localhost dockerd: time="2018-04-24T02:00:22.450621965Z" level=info msg="Container 0e9a27fd4092 failed to exit within 10 seconds of kill - trying direct SIGKILL"
```

Between "trying to kill container" and "calling sigkill" step it was possible to connect to the container. 

Seems similar to:
https://github.com/moby/moby/issues/35091
https://github.com/moby/moby/issues/30984

**Output of `docker version`:**

```
Docker version 17.06.2-ce, build cec0b72
```

**Output of `docker info`:**
Done on the different machine!
```
Containers: 11
 Running: 9
 Paused: 0
 Stopped: 2
Images: 14
Server Version: 17.06.2-ce
Storage Driver: devicemapper
 Pool Name: vg00-docker--pool
 Pool Blocksize: 524.3kB
 Base Device Size: 10.74GB
 Backing Filesystem: xfs
 Data file: 
 Metadata file: 
 Data Space Used: 6.145GB
 Data Space Total: 21.47GB
 Data Space Available: 15.33GB
 Metadata Space Used: 1.499MB
 Metadata Space Total: 218.1MB
 Metadata Space Available: 216.6MB
 Thin Pool Minimum Free Space: 2.147GB
 Udev Sync Supported: true
 Deferred Removal Enabled: true
 Deferred Deletion Enabled: true
 Deferred Deleted Device Count: 0
 Library Version: 1.02.107-RHEL7 (2015-10-14)
Logging Driver: json-file
Cgroup Driver: cgroupfs
Plugins: 
 Volume: local
 Network: bridge host macvlan null overlay
 Log: awslogs fluentd gcplogs gelf journald json-file logentries splunk syslog
Swarm: active
 NodeID: r0mhtl1jfcx56reradily0ukt
 Is Manager: true
 ClusterID: p7h1uf8hpslpyludj70a6pluv
 Managers: 3
 Nodes: 3
 Orchestration:
  Task History Retention Limit: 2
 Raft:
  Snapshot Interval: 10000
  Number of Old Snapshots to Retain: 0
  Heartbeat Tick: 1
  Election Tick: 3
 Dispatcher:
  Heartbeat Period: 5 seconds
 CA Configuration:
  Expiry Duration: 3 months
  Force Rotate: 0
 Root Rotation In Progress: false
 Node Address: 10.9.4.120
 Manager Addresses:
  10.9.4.120:2377
  10.9.4.213:2377
  10.9.4.60:2377
Runtimes: runc
Default Runtime: runc
Init Binary: docker-init
containerd version: 6e23458c129b551d5c9871e5174f6b1b7f6d1170
runc version: 810190ceaa507aa2727d7ae6f4790c76ec150bd2
init version: 949e6fa
Security Options:
 seccomp
  Profile: default
Kernel Version: 3.10.0-327.el7.x86_64
Operating System: CentOS Linux 7 (Core)
OSType: linux
Architecture: x86_64
CPUs: 3
Total Memory: 5.671GiB
Name: node-10-9-4-120
ID: P2ET:IHOQ:6OXW:4ZLJ:4WQW:H7RB:5EAV:UBY7:KNTL:Q453:CU2A:PBMJ
Docker Root Dir: /var/lib/docker
Debug Mode (client): false
Debug Mode (server): true
 File Descriptors: 187
 Goroutines: 362
 System Time: 2018-04-25T11:50:53.421489293Z
 EventsListeners: 10
Registry: https://index.docker.io/v1/
Experimental: false
Insecure Registries:
 docker-registry:5000
 127.0.0.0/8
Registry Mirrors:
 http://docker-registry:5001/
Live Restore Enabled: false

WARNING: bridge-nf-call-ip6tables is disabled
```

**Additional environment details (AWS, VirtualBox, physical, etc.):**
Happened with docker swarm deployed on 3 KVM VM's on CentOS 7.2.

**Comment by cpuguy83** [comment]
Thanks for reporting.
Docker 17.06-ce is nearly a year old and unsupported.
In addition, containerd and how Docker interacts with containerd is totally different now.

I'm going to close this since there's nothing really to investigate as this is reported against a very old codebase that has changed almost completely since, but feel free to discuss.
Thanks! 🙇 👼

**Comment by free2k** [comment]
@cpuguy83 I also have this problem. What is the cause? I am using version 1.13.1 now, is there any good way to fix this problem without upgrading docker? Because upgrading is more difficult for me.
